addons/cppcheckdata.py

```diff
@@ -1612,7 +1612,7 @@ def is_suppressed(location, message, errorId):
             return True
     return False
 
-def reportError(location, severity, message, addon, errorId, extra=''):
+def reportError(location, severity, message, addon, errorId, misra_severity='', extra=''):
```

Comment:
Spontanously it does not feel very good to add a "misra_severity" here. This is the generic function it's not related to misra. We have other addons also and I don't want to add various special parameters for those.

So well one approach could be to rename the parameters somehow to have generic parameters that works for all addons..

Reply:
I did not technically add the misra_severity, I just renamed the parameter, to the argument name used in misra.py, and then I re-used the extra parameter, for the extra information I wanted to add, since it seemed like a good name for additional error information.

https://github.com/danmar/cppcheck/blob/a87e9e104239785058e91811ea1dbf5bdc44e919/addons/misra.py#L4132

![image](https://github.com/danmar/cppcheck/assets/76603374/e53b431e-4f90-40f1-95ea-70941816c0ba)

Reply:
No it is not a good name for all the other addons.

If I report an error in the misc.py addon and then I want to provide some extra information to the error message then I don't feel that we should use `misra_severity`..

I want a generic argument name that will make sense in other addons also.